src/vs/workbench/contrib/files/browser/explorerService.ts

```diff
@@ -240,7 +250,8 @@ export class ExplorerService implements IExplorerService {
 		}
 
 		const fileStat = this.findClosest(resource);
-		if (fileStat) {
+		// If file or parent matches exclude patterns, do not reveal
+		if (fileStat && !this.matchesUpToRoot(fileStat)) {
```

Comment:
In some contexts the item should be revealed even if it's marked as autoreveal excluded. For instance when running the `revealInExplorer` command (`File: Reveal Active File in Sidebar`)

Reply:
Do you know why the reveal function accepts `boolean | string`?
It is passed to `view.selectResource()`, but that only ever checks if `reveal === true` and does not make use of the string.

Would it be ok to have the `revealInExplorer` command pass something like `force` for the reveal parameter and have the select function check for it?